### PR TITLE
py-enum-tools: add new package and dependencies

### PR DESCRIPTION
**Pull request by Barbozafab**
Ping @hyviquel .

### CODE REVIEW

**Review by tldahlgren — COMMENTED** [review]
Confirmed the homepage BUT `spack checksum` does not yield that version sha256.  How did you calculate it?

**Review by tldahlgren — CHANGES_REQUESTED** [review]
I'm getting a different sha256:

```
$  spack checksum py-enum-tools 0.10.0

==> Found 1 version of py-enum-tools:
  
  0.10.0  https://github.com/domdfcoding/enum_tools/archive/v0.10.0.tar.gz

==> Fetching https://github.com/domdfcoding/enum_tools/archive/v0.10.0.tar.gz

    version("0.10.0", sha256="049058d0c451bd05a73b449fd23e5314a86097d87cfcb17631ef89a977bc113c")
```

File: var/spack/repos/builtin/packages/py-enum-tools/package.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.package import *
+
+
+class PyEnumTools(PythonPackage):
+    """Tools to expand Python's enum module."""
+
+    homepage = "https://github.com/domdfcoding/enum_tools"
+    url = "https://github.com/domdfcoding/enum_tools/archive/v0.10.0.tar.gz"
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-enum-tools/package.py]: https://spack.readthedocs.io/en/latest/build_systems/pythonpackage.html#pypi-vs-github

File: var/spack/repos/builtin/packages/py-enum-tools/package.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+    version("0.10.0", sha256="777dc3cfb4314780bb9ca2460b518be58c3f29e13bd77b33badead4c2c136976")
+
+    depends_on("python@3.6:", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-enum-tools/package.py]: This line isn't needed, Spack only supports Python 3.7+ anyway.

Missing build dep on py-whey.

**Review by adamjstewart — CHANGES_REQUESTED** [review]

**Comment by Barbozafab** [comment]
> Confirmed the homepage BUT `spack checksum` does not yield that version sha256. How did you calculate it?

Woops! It seems I used the sha256 for the Pypi link, but added the package with the Github URL instead. I've just fixed it! Tyvm!

File: var/spack/repos/builtin/packages/py-apeye/package.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+class PyApeye(PythonPackage):
+    """Handy tools for working with URLs and APIs."""
+
+    homepage = "https://github.com/domdfcoding/apeye"
+    pypi = "apeye/apeye-1.4.1.tar.gz"
+
+    version("1.4.1", sha256="14ea542fad689e3bfdbda2189a354a4908e90aee4bf84c15ab75d68453d76a36")
+
+    depends_on("py-flit-core@3.2:4", type="build")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-apeye/package.py]:
```suggestion
    depends_on("py-flit-core@3.2:3", type="build")
```

File: var/spack/repos/builtin/packages/py-apeye/package.py [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+    depends_on("py-apeye-core@1.1.4:", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-apeye/package.py]:
```suggestion
    depends_on("py-apeye-core@1:", type=("build", "run"))
```

File: var/spack/repos/builtin/packages/py-consolekit/package.py [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+class PyConsolekit(PythonPackage):
+    """Additional utilities for click."""
+
+    homepage = "https://github.com/domdfcoding/consolekit"
+    pypi = "consolekit/consolekit-1.5.1.tar.gz"
+
+    version("1.5.1", sha256="55ea43e226863e1d618ec9b860c9842d84249d895c3376c03b158d8f3a335626")
+
+    depends_on("py-flit-core@3.2:4", type="build")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-consolekit/package.py]:
```suggestion
    depends_on("py-flit-core@3.2:3", type="build")
```

File: var/spack/repos/builtin/packages/py-consolekit/package.py [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+    depends_on("py-click@7.1.2:", type=("build", "run"))
+    depends_on("py-colorama@0.4.3:", type=("build", "run"), when="^python@:3.10 platform=windows")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-consolekit/package.py]:
```suggestion
    depends_on("py-colorama@0.4.3:", type=("build", "run"), when="^python@:3.9 platform=windows")
```

File: var/spack/repos/builtin/packages/py-consolekit/package.py [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+    depends_on("py-deprecation-alias@0.1.1:", type=("build", "run"))
+    depends_on("py-domdf-python-tools@2.6:", type=("build", "run"))
+    depends_on("py-mistletoe@0.7.2:", type=("build", "run"))
+    depends_on("py-typing-extensions@3.10:", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-consolekit/package.py:24]:
```suggestion
    depends_on("py-typing-extensions@3.10:", type=("build", "run"))
    conflicts("^py-typing-extensions@3.10.0.1")
```

File: var/spack/repos/builtin/packages/py-deprecation-alias/package.py [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+class PyDeprecationAlias(PythonPackage):
+    """Wrapper providing support for deprecated aliases."""
+
+    homepage = "https://github.com/domdfcoding/deprecation-alias"
+    pypi = "deprecation_alias/deprecation-alias-0.3.2.tar.gz"
+
+    version("0.3.2", sha256="1c9e1a5ddd0a276a1a18e7a4f9d56b53232217491c4549eaa45e51753013ce76")
+
+    depends_on("py-wheel@0.34.2:", type="build")
+    depends_on("py-setuptools@40.6:", type="build")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-deprecation-alias/package.py:19]:
```suggestion
    depends_on("py-setuptools@40.6:", type="build")
    conflicts("^py-setuptools@61")
```

File: var/spack/repos/builtin/packages/py-dist-meta/package.py [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+class PyDistMeta(PythonPackage):
+    """Parse and create Python distribution metadata."""
+
+    homepage = "https://github.com/repo-helper/dist-meta"
+    pypi = "dist_meta/dist-meta-0.8.0.tar.gz"
+
+    version("0.8.0", sha256="541d51f75b7f580c80d8d7b23112d0b4bf3edbc9442e425a7c4fcd75f4138551")
+
+    depends_on("py-wheel@0.34.2:", type="build")
+    depends_on("py-setuptools@40.6:", type="build")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-dist-meta/package.py:19]:
```suggestion
    depends_on("py-setuptools@40.6:", type="build")
    conflicts("^py-setuptools@61")
```

File: var/spack/repos/builtin/packages/py-dom-toml/package.py [hunk 10]
```diff
@@ -0,0 +1,20 @@
...
+
+class PyDomToml(PythonPackage):
+    """Dom's tools for Tom's Obvious, Minimal Language."""
+
+    homepage = "https://github.com/domdfcoding/dom_toml"
+    pypi = "dom_toml/dom_toml-0.6.1.tar.gz"
+
+    version("0.6.1", sha256="a0bfc204ae32c72ed36e526dce56108a3b20741ac3c055207206ce3b2f302868")
+
+    depends_on("py-flit-core@3.2:4", type="build")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-dom-toml/package.py]:
```suggestion
    depends_on("py-flit-core@3.2:3", type="build")
```

File: var/spack/repos/builtin/packages/py-domdf-python-tools/package.py [hunk 11]
```diff
@@ -0,0 +1,23 @@
...
+class PyDomdfPythonTools(PythonPackage):
+    """Helpful functions for Python"""
+
+    homepage = "https://github.com/domdfcoding/domdf_python_tools"
+    pypi = "domdf_python_tools/domdf_python_tools-3.6.1.tar.gz"
+
+    version("3.6.1", sha256="acc04563d23bce4d437dd08af6b9bea788328c412772a044d8ca428a7ad861be")
+
+    depends_on("py-wheel@0.34.2:", type="build")
+    depends_on("py-setuptools@40.6:", type="build")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-domdf-python-tools/package.py:19]:
```suggestion
    depends_on("py-setuptools@40.6:", type="build")
    conflicts("^py-setuptools@61")
```

File: var/spack/repos/builtin/packages/py-domdf-python-tools/package.py [hunk 12]
```diff
@@ -0,0 +1,23 @@
...
+    depends_on("py-importlib-metadata@3.6:", type=("build", "run"), when="^python@:3.9")
+    depends_on("py-importlib-resources@3:", type=("build", "run"), when="^python@:3.7")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-domdf-python-tools/package.py]:
```suggestion
    depends_on("py-importlib-metadata@3.6:", type=("build", "run"), when="^python@:3.8")
```

File: var/spack/repos/builtin/packages/py-handy-archives/package.py [hunk 13]
```diff
@@ -0,0 +1,18 @@
...
+
+class PyHandyArchives(PythonPackage):
+    """Some handy archive helpers for Python."""
+
+    homepage = "https://github.com/domdfcoding/handy-archives"
+    pypi = "handy_archives/handy_archives-0.2.0.tar.gz"
+
+    version("0.2.0", sha256="fba21101fd9e29d5e3b72823261aaae06b9350686f0d2067786d64dce73eb3f6")
+
+    depends_on("py-flit-core@3.2:4", type="build")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-handy-archives/package.py]:
```suggestion
    depends_on("py-flit-core@3.2:3", type="build")
```

File: var/spack/repos/builtin/packages/py-mistletoe/package.py [hunk 14]
```diff
@@ -0,0 +1,19 @@
...
+class PyMistletoe(PythonPackage):
+    """A fast, extensible Markdown parser in pure Python."""
+
+    homepage = "https://github.com/miyuchina/mistletoe"
+    pypi = "mistletoe/mistletoe-1.2.1.tar.gz"
+
+    version("1.2.1", sha256="7d0c1ab3747047d169f9fc4b925d1cba3f5c13eaf0b90c365b72e47e59d00a02")
+
+    depends_on("py-setuptools", type="build")
+    depends_on("py-pygments@2.11.2:2.16", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-mistletoe/package.py]: Where does pygments come from?

File: var/spack/repos/builtin/packages/py-mistletoe/package.py [hunk 15]
```diff
@@ -0,0 +1,19 @@
...
+
+
+class PyMistletoe(PythonPackage):
+    """A fast, extensible Markdown parser in pure Python."""
+
+    homepage = "https://github.com/miyuchina/mistletoe"
+    pypi = "mistletoe/mistletoe-1.2.1.tar.gz"
+
+    version("1.2.1", sha256="7d0c1ab3747047d169f9fc4b925d1cba3f5c13eaf0b90c365b72e47e59d00a02")
+
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-mistletoe/package.py:17]:
```python
depends_on("python@3.5:3", type=("build", "run"))
```

File: var/spack/repos/builtin/packages/py-pyproject-parser/package.py [hunk 16]
```diff
@@ -0,0 +1,29 @@
...
+class PyPyprojectParser(PythonPackage):
+    """Parser for 'pyproject.toml'"""
+
+    homepage = "https://github.com/repo-helper/pyproject-parser"
+    pypi = "pyproject_parser/pyproject-parser-0.9.1.tar.gz"
+
+    version("0.9.1", sha256="fa0b2ff78bc95788b08d00e1aafa66d3f7f3ab693f19d9c2e23e20000a69fd9b")
+
+    depends_on("py-wheel@0.34.2:", type="build")
+    depends_on("py-setuptools@40.6:", type="build")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pyproject-parser/package.py:19]:
```suggestion
    depends_on("py-setuptools@40.6:", type="build")
    conflicts("^py-setuptools@61")
```

File: var/spack/repos/builtin/packages/py-pyproject-parser/package.py [hunk 17]
```diff
@@ -0,0 +1,29 @@
...
+    depends_on("py-apeye-core@1:", type=("build", "run"))
+    depends_on("py-attrs@20.3:", type=("build", "run"))
+    depends_on("py-dom-toml@0.4:", type=("build", "run"))
+    depends_on("py-domdf-python-tools@2.8:", type=("build", "run"))
+    depends_on("py-natsort@7.1.1:", type=("build", "run"))
+    depends_on("py-packaging@20.9:", type=("build", "run"))
+    depends_on("py-shippinglabel@1:", type=("build", "run"))
+    depends_on("py-toml@0.10.2:", type=("build", "run"))
+    depends_on("py-tomli@1.2.3:", type=("build", "run"), when="^python@:3.11")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pyproject-parser/package.py]:
```suggestion
    depends_on("py-tomli@1.2.3:", type=("build", "run"), when="^python@:3.10")
```

File: var/spack/repos/builtin/packages/py-pyproject-parser/package.py [hunk 18]
```diff
@@ -0,0 +1,29 @@
...
+    depends_on("py-typing-extensions@3.7.4.3:", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pyproject-parser/package.py:30]:
```suggestion
    depends_on("py-typing-extensions@3.7.4.3:", type=("build", "run"))
    conflicts("^py-typing-extensions@4.7.0")
```

File: var/spack/repos/builtin/packages/py-shippinglabel/package.py [hunk 19]
```diff
@@ -0,0 +1,26 @@
...
+class PyShippinglabel(PythonPackage):
+    """Utilities for handling packages."""
+
+    homepage = "https://github.com/domdfcoding/shippinglabel"
+    pypi = "shippinglabel/shippinglabel-1.5.0.tar.gz"
+
+    version("1.5.0", sha256="b2332bf29853e29f877dab21f17f2a8632fd2b48f5f32a6fa0b254a6fdd0a451")
+
+    depends_on("py-wheel@0.34.2:", type="build")
+    depends_on("py-setuptools@40.6:", type="build")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-shippinglabel/package.py:19]:
```suggestion
    depends_on("py-setuptools@40.6:", type="build")
    conflicts("^py-setuptools@61")
```

File: var/spack/repos/builtin/packages/py-whey/package.py [hunk 20]
```diff
@@ -0,0 +1,23 @@
...
+class PyWhey(PythonPackage):
+    """A simple Python wheel builder for simple projects."""
+
+    homepage = "https://github.com/repo-helper/whey"
+    pypi = "whey/whey-0.0.24.tar.gz"
+
+    version("0.0.24", sha256="411905d85aa8aa239733818894e08dc20b682f0a3614f942aa35b430db568aa2")
+
+    depends_on("py-wheel@0.34.2", type="build")
+    depends_on("py-setuptools@40.6:", type="build")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-whey/package.py:19]:
```suggestion
    depends_on("py-setuptools@40.6:", type="build")
    conflicts("^py-setuptools@61")
```

File: var/spack/repos/builtin/packages/py-whey/package.py [hunk 21]
```diff
@@ -0,0 +1,23 @@
...
+    depends_on("py-consolekit@1.4.1:", type=("build", "run"))
+    depends_on("py-dist-meta@0.1:", type=("build", "run"))
+    depends_on("py-handy-archives@0.2:", type=("build", "run"))
+    depends_on("py-pyproject-parser@0.6:", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-whey/package.py:31]: Missing a bunch of deps:
```
click>=7.1.2
consolekit>=1.4.1
dist-meta>=0.1.0
dom-toml>=0.4.0
domdf-python-tools>=2.8.0
handy-archives>=0.1.0
natsort>=7.1.1
packaging>=20.9
pyproject-parser>=0.6.0
shippinglabel>=0.16.0
```

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-mistletoe/package.py (hunk 14 above)
Comment (Barbozafab) [reply var/spack/repos/builtin/packages/py-mistletoe/package.py]: It was specified in a `requirements.txt` but I just realised the file was not used by `setuptools`. I removed the dependency.

**Review by tldahlgren — DISMISSED** [review]
Reconfirmed all affected homepages and version sha256.  Will defer to @adamjstewart for final approval.

**Comment by tldahlgren** [comment]
This PR is suffering from a CI issue I've seen in a number of other PRs.  Hopefully the fix PR is merged soon so this one can proceed.

File: var/spack/repos/builtin/packages/py-apeye/package.py [hunk 22]
```diff
@@ -0,0 +1,21 @@
...
+class PyApeye(PythonPackage):
+    """Handy tools for working with URLs and APIs."""
+
+    homepage = "https://github.com/domdfcoding/apeye"
+    pypi = "apeye/apeye-1.4.1.tar.gz"
+
+    version("1.4.1", sha256="14ea542fad689e3bfdbda2189a354a4908e90aee4bf84c15ab75d68453d76a36")
+
+    depends_on("py-flit-core@3.2:3", type="build")
+    depends_on("py-apeye-core@1:", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-apeye/package.py:19]: Missing dep: `domdf-python-tools>=2.6.0`

**Review by adamjstewart — CHANGES_REQUESTED** [review]

**Review by adamjstewart — APPROVED** [review]

**Comment by adamjstewart** [comment]
Thanks for adding all of these packages!

**Pull request merged**